@Test public void testAngle() throws MathArithmeticException {
  Assert.assertEquals(0.22572612855273393616,FieldVector3D.angle(createVector(1,2,3,3),createVector(4,5,6,3)).getReal(),1.0e-12);
  Assert.assertEquals(7.98595620686106654517199e-8,FieldVector3D.angle(createVector(1,2,3,3),createVector(2,4,6.000001,3)).getReal(),1.0e-12);
  Assert.assertEquals(3.14159257373023116985197793156,FieldVector3D.angle(createVector(1,2,3,3),createVector(-2,-4,-6.000001,3)).getReal(),1.0e-12);
  try {
    FieldVector3D.angle(createVector(0,0,0,3),createVector(1,0,0,3));
    Assert.fail("an exception should have been thrown");
  }
 catch (  MathArithmeticException ae) {
  }
}
